MathLib/LinAlg/PETSc/PETScVector.cpp

```diff
@@ -156,6 +164,27 @@ void PETScVector::getGlobalVector(PetscScalar u[])
 #endif
 }
 
+void PETScVector::setLocalAccessibleVector() const
+{
+    if (_entry_array.size() == 0)
+    {
+        const PetscInt array_size
+            = _global_ids2local_ids_ghost.size() > 0 ?
+              _size_loc + _size_ghosts: _size;
+        _entry_array.resize(array_size);
+    }
+
+    if (_global_ids2local_ids_ghost.size() > 0)
```

Comment:
Changed to !_global_ids2local_ids_ghost.empty().
@TomFischer  will perform clang-format in another PR.